File: 125/solution.c
#include "../tools/pe.h"

gint
main()
{
    glong max, sum, count = 0;
    GArray *squares;
    GArray *pals;
    gint i;
    gdouble square_root, square;

    max = pow(10, 3);
    square_root = sqrt(max);
    squares = g_array_new(FALSE, FALSE, sizeof (gint));
    pals = g_array_new(FALSE, FALSE, sizeof (gint));

    for(i = 1; i <= square_root; i++)
    {
        square = pow(i, 2);
        printf("%0.f\n", square);
        g_array_append_val(squares, square);
    }

    printf("%f ", &g_array_index(squares, gdouble, 0));


    return 0;
}

File: 125/solution.rb
#!/usr/bin/env ruby
#
# projecteuler.net
#
# problem 125
#
# The palindromic number 595 is interesting because it can be written as
# the sum of consecutive squares: 6^2 + 7^2 + 8^2 + 9^2 + 10^2 + 11^2 + 12^2.
#
# There are exactly eleven palindromes below one-thousand that can be
# written as consecutive square sums, and the sum of these palindromes
# is 4164. Note that 1 = 0^2 + 1^2 has not been included as this problem
# is concerned with the squares of positive integers.
#
# Find the sum of all the numbers less than 10^8 that are both
# palindromic and can be written as the sum of consecutive squares.
#
# wrong answer: 2916867073

require '../tools/ffi_pe'

include PEMethods

max = 10**8
squares = []
pals = []
sum = 0
count = 0

1.upto(Math.sqrt(max)) do |i|
    squares << i ** 2
end

i = 0
j = i

while i <= squares.length - 2 do
    sum = squares[i..j].inject(:+)
    if is_palindrome(sum) and sum != 1 and sum <= max
        pals << sum
    end
    j += 1
    if j == squares.length
        i += 1
        j = i + 1
    end
end
#puts pals.inspect
#puts pals.size
puts pals.inject(:+)



